### PR TITLE
feat: Add timeseries endpoint, closes #124

### LINKED ISSUE #124: Add timeseries endpoint

### PR DESCRIPTION
**Pull request by KitBurgess**

### CODE REVIEW

File: vortexasdk/client.py [hunk 1]
```diff
@@ -75,10 +79,10 @@ def search(self, resource: str, **data) -> List:
...
         flattened = [x for y in responses for x in y]
 
-        assert len(flattened) == total, (
-            f"Incorrect number of records returned from API. "
-            f"Actual: {len(flattened)}, expected: {total}"
-        )
+        # assert len(flattened) == total, (
+        #     f"Incorrect number of records returned from API. "
+        #     f"Actual: {len(flattened)}, expected: {total}"
+        # )
```
Comment (cvonsteg) [inline vortexasdk/client.py]: Why is this commented out?

File: vortexasdk/endpoints/timeseries_result.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class TimeSeriesResult(Result):
+    """Container class that holds the result obtained from calling a time series endpoint."""
+
+    def to_list(self) -> List[TimeSeriesItem]:
+        """Represents time series as a list."""
+        list_of_dicts = super().to_list()
+        with Pool(os.cpu_count()) as pool:
+            return list(pool.map(TimeSeriesItem.from_dict, list_of_dicts))
+
+    def to_df(self, columns=None) -> pd.DataFrame:
```
Comment (cvonsteg) [inline vortexasdk/endpoints/timeseries_result.py]: `columns` arg never appears to be used.  Is it required here?

Comment (cvonsteg) [reply vortexasdk/endpoints/timeseries_result.py]: I can see the base class abstract method has it in its signature, is it worth adding it to the return statement's signature?

**Comment by KitBurgess** [comment]
Will merge this so that clients can make use of it, then address @cvonsteg 's points above.

**Pull request merged**